test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -54,15 +55,21 @@ const { ESMLoader } = esmLoaderModule;
     // This doesn't matter (just to avoid errors)
     return {
       format: 'module',
+      shortCircuit: true,
       source: '',
     };
   }
 
-  const customLoader = {
-    // Ensure ESMLoader actually calls the custom hooks
-    resolve: mustCall(resolve),
-    load: mustCall(load),
-  };
+  const customLoader = [
+    {
+      exports: {
+        // Ensure ESMLoader actually calls the custom hooks
+        resolve: mustCall(resolve),
+        load: mustCall(load),
+      },
+      url: 'node:test/es-module/test-esm-loader-hooks.mjs',
```

Comment:
nit: using a `node:` scheme doesn't really apply here. Any reason not to use the `file:` protocol?
```suggestion
      url: import.meta.url,
```